src/main/java/org/entando/kubernetes/model/bundle/processor/PluginProcessor.java

```diff
@@ -148,8 +151,23 @@ private void logDescriptorWarnings(PluginDescriptor descriptor) {
         }
     }
 
+    private PluginDescriptor ensurePluginDescriptorVersionIsSet(PluginDescriptor descriptor) {
+        if (StringUtils.isEmpty(descriptor.getDescriptorVersion())) {
+            Integer intVersion = BundleUtilities.getPluginDescriptorIntegerVersion(descriptor);
+            descriptor.setDescriptorVersion(BundleUtilities.composePluginDescriptorVersion(intVersion));
+        }
+
+        return descriptor;
+    }
+
     private void validateDescriptorOrThrow(PluginDescriptor descriptor) {
 
+        // validate version
+        Matcher matcher = Pattern.compile(BundleUtilities.PLUGIN_DESCRIPTOR_VERSION_REGEXP).matcher(descriptor.getDescriptorVersion());
+        if (!matcher.matches()) {
+            throw new InvalidBundleException(String.format(VERSION_NOT_VALID, descriptor.getComponentKey()));
```

Comment:
There should be logging to go with this. Even if the exception is logged elsewhere.